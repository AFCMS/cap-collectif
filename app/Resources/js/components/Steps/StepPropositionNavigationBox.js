// @flow
import * as React from 'react';
import styled from 'styled-components';
import { graphql, QueryRenderer } from 'react-relay';
import type { RelayGlobalId } from '../../types';
import colors from '../../utils/colors';
import environment, { graphqlError } from '../../createRelayEnvironment';
import type {
  StepPropositionNavigationBoxQueryResponse,
  StepPropositionNavigationBoxQueryVariables,
} from '~relay/StepPropositionNavigationBoxQuery.graphql';
import StepPropositionNavigation from './StepPropositionNavigation';

export type Props = {|
  +stepId: RelayGlobalId,
  +relatedSlug: string,
|}

export const STEP_PROPOSITION_NAVIGATION_HEIGHT = 100;

const StepPropositionNavigationBoxInner = styled.div`
  height: ${STEP_PROPOSITION_NAVIGATION_HEIGHT}px;
  max-height: ${STEP_PROPOSITION_NAVIGATION_HEIGHT}px;
  min-height: ${STEP_PROPOSITION_NAVIGATION_HEIGHT}px;
  padding: 40px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  & h2 {
    color: ${colors.white};
    margin: 0;    
  }
`;

const renderStepPropositionNavigation = (
  {
    error,
    props,
  }: {
    ...ReactRelayReadyState,
    props: ?StepPropositionNavigationBoxQueryResponse,
  },
) => {
  if (error) {
    console.log(error); // eslint-disable-line no-console
    return graphqlError;
  }
  if (props) {
    if (props.step) {
      const { step } = props;
      return (
        // $FlowFixMe
        <StepPropositionNavigation step={step}/>
      );
    }
    return graphqlError;
  }
  return null;
};

const StepPropositionNavigationBox = ({ stepId, relatedSlug }: Props) => {
  return (
    <StepPropositionNavigationBoxInner>
      <QueryRenderer
        environment={environment}
        query={graphql`
            query StepPropositionNavigationBoxQuery(
              $stepId: ID!
              $relatedSlug: String!
            ) {
              step: node(id: $stepId) {
                ...StepPropositionNavigation_step @arguments(relatedSlug: $relatedSlug)
              }
            }
          `}
        variables={
          ({
            stepId,
            relatedSlug
          }: StepPropositionNavigationBoxQueryVariables)
        }
        render={renderStepPropositionNavigation}
      />
    </StepPropositionNavigationBoxInner>
  );
};

export default StepPropositionNavigationBox;
