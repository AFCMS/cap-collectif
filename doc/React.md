### How to write localizable components

Just import the appropriate [component](https://github.com/yahoo/react-intl/wiki#the-react-intl-module) from `react-intl`

- For localizable text use
[`<FormattedMessage>`](https://github.com/yahoo/react-intl/wiki/Components#formattedmessage).

- For date and time:
[`<FormattedDate>`](https://github.com/yahoo/react-intl/wiki/Components#formatteddate)
[`<FormattedTime>`](https://github.com/yahoo/react-intl/wiki/Components#formattedtime)
[`<FormattedRelative>`](https://github.com/yahoo/react-intl/wiki/Components#formattedrelative)

- For numbers and currencies:
[`<FormattedNumber>`](https://github.com/yahoo/react-intl/wiki/Components#formattednumber)
[`<FormattedPlural>`](https://github.com/yahoo/react-intl/wiki/Components#formattedplural)

- If possible, do not use `<FormattedHTMLMessage>`, see how to use *Rich Text Formatting* with
[`<FormattedMessage>`](https://github.com/yahoo/react-intl/wiki/Components#formattedmessage)

- When you need an imperative formatting API, use the [`injectIntl`](https://github.com/yahoo/react-intl/wiki/API#injectintl) High-Order Component.
