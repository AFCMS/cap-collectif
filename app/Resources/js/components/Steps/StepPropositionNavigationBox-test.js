// @flow
/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import { StepPropositionNavigationBox } from './StepPropositionNavigationBox'

describe('<StepPropositionNavigationBox />', () => {

  it('should render correctly', () => {

    const props = {
      stepId: 'cstep1',
      relatedSlug: 'deuxieme-consultation'
    };

    const wrapper = shallow(<StepPropositionNavigationBox {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });

});
