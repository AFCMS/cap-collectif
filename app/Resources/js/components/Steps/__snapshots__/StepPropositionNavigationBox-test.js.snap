// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepPropositionNavigationBox /> should render correctly 1`] = `
<StepPropositionNavigationBox__StepPropositionNavigationBoxInner>
  <ReactRelayQueryRenderer
    environment={"RelayModernEnvironment()"}
    query={[Function]}
    render={[Function]}
    variables={
      Object {
        "relatedSlug": "deuxieme-consultation",
        "stepId": "cstep1",
      }
    }
  />
</StepPropositionNavigationBox__StepPropositionNavigationBoxInner>
`;
